### PR TITLE
Write Lighthouse reports to a new *_lighthouse table

### PR DESCRIPTION
**Pull request by rviscomi**
To summarize the latest changes in this PR:

- LH reports are no longer written to the *_pages table
- reportCategories..audits are omitted
- screenshot-thumbnails items (base64 encoded images) are omitted
- report is `NULL` rather than the string `"null"` when applicable

These are efforts to minimize the row size of the LH report so that we can fit more reports. Here are the number of skipped rows per optimization:

- none: ~12k
- omit audits: ~4.8k
- ^ + omit images: ~4.6k
- ^ + separate table: ~4.6k

### CODE REVIEW

**Comment by igrigorik** [comment]
@rviscomi can you share an example of what these duplicates are?

**Comment by rviscomi** [comment]
@igrigorik here's the same color-contrast audit in two places

image

**Comment by igrigorik** [comment]
Hmm, @ebidel why is this data duplicated in LH reports? I imagine optimizing report size is not a high on the agenda for LH, but duplication still feels a bit odd.

**Comment by ebidel** [comment]
Known issue: https://github.com/GoogleChrome/lighthouse/issues/1999

**Comment by igrigorik** [comment]
Gotcha, thanks Eric. Would be great to get this fixed upstream.. With more and more tools integrating LH reporting, I imagine this will become a recurring pain point.

**Comment by rviscomi** [comment]
**Dropped rows**
Before pruning: 12,121
After pruning: 4,882

**Comment by ebidel** [comment]
I'm going to start the work upstream to remove the dupes. No promises on timeline though :)

**Comment by rviscomi** [comment]
Updated to use a separate HAR table. Doesn't seem to have an effect on the number of skipped rows, surprisingly. Also, ~50k pages have `NULL` reports. The 6/15 table without the changes in this PR has only ~33k `"null"` reports.

One positive side-effect of this change is that the processing takes the normal amount of 45 minutes, rather than 2.5 hours.

image

**Comment by igrigorik** [comment]
Faster import time is a nice win. Looks like we'll need to find a way to prune the data a bit more? =/

**Comment by ebidel** [comment]
FWIW, I see only 18 tables with non-`null` repots for 6/20:

```
SELECT
  count(*)
FROM
  httparchive.har.2017_06_20_android_lighthouse
WHERE
  report != 'null'
```

**Comment by rviscomi** [comment]
Sorry you can ignore the 6/20 tables. Those are temporary and just for testing.

**Comment by rviscomi** [comment]
Here's a list of URLs and ranks for sites omitted from the LH table, presumably because they were too large and skipped: [query](https://bigquery.cloud.google.com/savedquery/226352634162:cf31392dc61a4592a1c16a41df395410) ([results](https://docs.google.com/spreadsheets/d/1oYHPX7wGSZVnDtbBXLbPqEZJWgmrahT_1Woked_7hW0/edit?usp=sharing))

Lowest rank is 250 so it does affect some of the more popular sites (eg slate, wsj, digg, theknot, etc).

Using this data, I can analyze these specific HARs for ways to minimize bytes.

**Comment by rviscomi** [comment]
@igrigorik does this LGTY? It'd be good to standardize the schema for a new LH table for the 7/15 crawl ending soon. I can follow up with optimizations to retain as many reports as possible.

File: dataflow/java/src/main/java/com/httparchive/dataflow/BigQueryImport.java [hunk 1]
```diff
@@ -160,20 +162,40 @@ public void processElement(ProcessContext c) {
...
-                        .set("payload", pageJSON)
-                        .set("lighthouse", lighthouseJSON);
+                        .set("report", lighthouseJSON);
 
-                String pageRowJSON = MAPPER.writeValueAsString(pageRow);
-                Integer pageRowSize = pageRowJSON.getBytes("UTF-8").length;
-                if (pageRowSize > MAX_CONTENT_SIZE) {
+                String lighthouseRowJSON = MAPPER.writeValueAsString(lighthouseRow);
+                Integer lighthouseRowSize = lighthouseRowJSON.getBytes("UTF-8").length;
+                if (lighthouseRowSize > MAX_CONTENT_SIZE) {
```
Comment (igrigorik) [inline dataflow/java/src/main/java/com/httparchive/dataflow/BigQueryImport.java]: Should you account for URL size as well? We do this for body table.

Comment (rviscomi) [inline dataflow/java/src/main/java/com/httparchive/dataflow/BigQueryImport.java]: Done

**Comment by rviscomi** [comment]
Updated production dataflow worker

**Comment by ebidel** [comment]
Rick, when do you expect the first dumps to show up?

**Comment by rviscomi** [comment]
@ebidel The 6/1, 6/15, and 7/1 crawls have been beta testing this schema so it's already in prod. The 7/15 crawl looks just about done so it should also be processed in the next few days.

**Pull request merged**